### DISCUSSION TITLE
License of Kornia library

### DISCUSSION THREAD

**Discussion by JANGSOONMYUN** [kornia/kornia#2646] [Questions] [answered]
According to the Kornia license policy, it is allowed to use the library for commercial purposes. However, I have noticed that the LoFTR module includes some code from the SuperGluePretrainedNetwork, which does not permit commercial use. (LoFTR requires https://raw.githubusercontent.com/magicleap/SuperGluePretrainedNetwork/master/models/utils.py)

But this problem originates from the LoFTR source code, not from the Kornia library itself. I am not certain whether Kornia uses SuperGluePretrainedNetwork code or not.

So, I would like to know if I can use any modules, especially LoFTR and LightGlue, for commercial purposes.

**Comment by johnnv1** [comment]
@ducha-aiki any idea about it?

**Comment by ducha-aiki** [comment, marked as answer]
Hi,
Even the original LoFTR doesn’t really depend on the code you shared, they just use it for evaluation script.
The LoFTR in kornia does not include any of SuperGlue code, so it is free.

**Reply by JANGSOONMYUN** [reply]
Thank you! So I just need to remove the code from LoFTR code, right?

**Reply by ducha-aiki** [reply]
@JANGSOONMYUN or use LoFTR directly from kornia :)

**Comment by edgarriba** [comment]
To add on @ducha-aiki reply, our implementation is based on: https://github.com/zju3dv/LoFTR which is licenced under Apache2.

**Reply by JANGSOONMYUN** [reply]
Yes that was the reason I decided to use it. But the utils.py file from SuperGlue made me confused if LoFTR cannot be used for commercial purposes. Thank you.

**Reply by edgarriba** [reply]
i would say if you don't ship those files would be fine, but consult with an OSS specialized lawyer.
